rfc/rfc-0006-child-resource-names.md

```diff
@@ -0,0 +1,43 @@
+# RFC-0006: Child Resource Names
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1369](https://github.com/projectriff/riff/pull/1369)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When a controller creates a new resource, it need to either pick a static name, or let kubernetes generate a name. Static names are predictable, but may already be in use. Generated names cannot be predicated, but are guaranteed to be available.
+
+### Anti-Goals
+This RFC does not prescribe which naming scheme to use for any specific resource.
+
+## Solution
+While there will not be a cut and dry answer for each situation, there are heuristics to guide the decision:
+- if a resource is not consumed directly by a non-controlled resource by name, generate the name (e.g. Pods in a ReplicaSet)
+- if the resource can be queried via a label selector instead of by name, generate the name (e.g. Pods fronted by a Service)
```

Comment:
I'm not sure this is always advisable. For instance, if a label selector doesn't match a unique resource, then it may be necessary to use a static name.

Reply:
List queries are inherently able to return zero-to-many resources. It may still be desirable to use a listing query to find a resource even if only a single resource is valid.

For example, our controllers often list resources controlled by the resource being reconciled to find the child. We expect either zero or one object to exist, but if multiple objects exist then we can handle that by deleting the superfluous object.